plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -243,6 +253,676 @@ public void testFlushPartitionCache()
         computeActual(format("DROP TABLE %s", fullyQualifiedTestTableName));
     }
 
+    @Test
+    public void testPartitionProjectionOnVarCharColumnWithEnumProjectionAndStorageLocationTemplate()
+    {
+        String tableName = "nation_" + randomTableSuffix();
+        String fullyQualifiedTestTableName = getTestTableName(tableName);
```

Comment:
> If a projected partition does not exist in Amazon S3, Athena will still project the partition.

Could you add a test for this usecase as well?

Reply:
It's a default behaviour of Trino Hive Connector. When we point to non existing FS resource we just ignore it.
Thats why i did not add this test case.

Reply:
`hive.ignore-absent-partitions` defaults to `false` AFAIK. We do ignore "empty" files when reading though.

Reply:
I will test it again. But my observations where that even if I project un-existing partition nothing happens.
I mean there is not exception. Let me double check that.